src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -60,6 +62,28 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool)(implicit settings: ErgoS
     new ErgoMemPool(pool.invalidate(tx))
   }
 
+  // Check if transaction is double-spending inputs spent in the mempool.
+  // If so, the new transacting is replacing older ones if it is paying more than all of them.
+  // Otherwise, the new transaction being rejected.
+  private def acceptIfNoDoubleSpend(tx: ErgoTransaction): (ErgoMemPool, ProcessingOutcome) = pool.synchronized {
```

Comment:
This `synchronized` is suspicious. Why it is necessary? Considering pool is immutable, and this method creates a new instance of it, it looks like the syncronization should lock on other object. Probably on a higher level of ErgoNodeViewHolder.txModify method.

Reply:
removed